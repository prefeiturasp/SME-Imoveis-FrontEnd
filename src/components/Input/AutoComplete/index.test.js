import { mount } from "enzyme";
import React from "react";
import { AutoComplete } from "./index";

describe("test <AutoComplete>", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(<AutoComplete value="" input={{ value: "" }} />);
  });

  it("renders component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
