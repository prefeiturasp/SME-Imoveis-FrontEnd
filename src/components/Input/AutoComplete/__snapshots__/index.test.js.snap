// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <AutoComplete> renders component 1`] = `
<AutoComplete
  className=""
  disabled={false}
  esconderAsterisco={false}
  handleChange={[Function]}
  helpText=""
  input={
    Object {
      "value": "",
    }
  }
  label=""
  labelClassName=""
  meta={Object {}}
  name=""
  placeholder=""
  required={false}
  value=""
>
  <div
    className="input"
  >
    <AutoSuggestAddress
      className=" undefined"
      onAddressSelected={[Function]}
      required={false}
      value=""
    >
      <Autosuggest
        alwaysRenderSuggestions={false}
        focusInputOnSuggestionClick={true}
        getSuggestionValue={[Function]}
        highlightFirstSuggestion={false}
        id="1"
        inputProps={
          Object {
            "autoFocus": true,
            "className": "form-control  undefined",
            "name": undefined,
            "onChange": [Function],
            "placeholder": "Ex: Rua Doutor Diogo de Faria",
            "required": false,
            "type": "search",
            "value": "",
          }
        }
        multiSection={false}
        onSuggestionsClearRequested={[Function]}
        onSuggestionsFetchRequested={[Function]}
        renderSuggestion={[Function]}
        renderSuggestionsContainer={[Function]}
        shouldRenderSuggestions={[Function]}
        suggestions={Array []}
        theme={
          Object {
            "container": "react-autosuggest__container",
            "containerOpen": "react-autosuggest__container--open",
            "input": "react-autosuggest__input",
            "inputFocused": "react-autosuggest__input--focused",
            "inputOpen": "react-autosuggest__input--open",
            "sectionContainer": "react-autosuggest__section-container",
            "sectionContainerFirst": "react-autosuggest__section-container--first",
            "sectionTitle": "react-autosuggest__section-title",
            "suggestion": "react-autosuggest__suggestion",
            "suggestionFirst": "react-autosuggest__suggestion--first",
            "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
            "suggestionsContainer": "react-autosuggest__suggestions-container",
            "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
            "suggestionsList": "react-autosuggest__suggestions-list",
          }
        }
      >
        <Autowhatever
          containerProps={Object {}}
          getSectionItems={[Function]}
          highlightedItemIndex={null}
          highlightedSectionIndex={null}
          id="1"
          inputProps={
            Object {
              "autoFocus": true,
              "className": "form-control  undefined",
              "name": undefined,
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "onKeyDown": [Function],
              "placeholder": "Ex: Rua Doutor Diogo de Faria",
              "required": false,
              "type": "search",
              "value": "",
            }
          }
          itemProps={[Function]}
          items={Array []}
          multiSection={false}
          renderInputComponent={[Function]}
          renderItem={[Function]}
          renderItemData={
            Object {
              "query": "",
            }
          }
          renderItemsContainer={[Function]}
          renderSectionTitle={[Function]}
          theme={
            Object {
              "container": "react-autosuggest__container",
              "containerOpen": "react-autosuggest__container--open",
              "input": "react-autosuggest__input",
              "inputFocused": "react-autosuggest__input--focused",
              "inputOpen": "react-autosuggest__input--open",
              "item": "react-autosuggest__suggestion",
              "itemFirst": "react-autosuggest__suggestion--first",
              "itemHighlighted": "react-autosuggest__suggestion--highlighted",
              "itemsContainer": "react-autosuggest__suggestions-container",
              "itemsContainerOpen": "react-autosuggest__suggestions-container--open",
              "itemsList": "react-autosuggest__suggestions-list",
              "sectionContainer": "react-autosuggest__section-container",
              "sectionContainerFirst": "react-autosuggest__section-container--first",
              "sectionTitle": "react-autosuggest__section-title",
            }
          }
        >
          <div
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-1"
            className="react-autosuggest__container"
            key="react-autowhatever-1-container"
            role="combobox"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls="react-autowhatever-1"
              autoComplete="off"
              autoFocus={true}
              className="form-control  undefined"
              key="react-autowhatever-1-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Ex: Rua Doutor Diogo de Faria"
              required={false}
              type="search"
              value=""
            />
            <div
              className="react-autosuggest__suggestions-container"
              id="react-autowhatever-1"
              key="react-autowhatever-1-items-container"
              role="listbox"
            />
          </div>
        </Autowhatever>
      </Autosuggest>
    </AutoSuggestAddress>
    <HelpText
      helpText=""
    >
      <div
        className="help-text"
      />
    </HelpText>
    <InputErroMensagem
      meta={Object {}}
    >
      <div
        className="error-or-warning-message"
      />
    </InputErroMensagem>
  </div>
</AutoComplete>
`;
